### PR TITLE
feat: dokku_checks library

### LINKED ISSUE #135: Disabling zero downtime deployments for an app
What would be the recommended way to disable the CHECKS functionality of dokku?

### PR DESCRIPTION
**Pull request by fr3fou**
Closes #135

### CODE REVIEW

**Comment by ltalirz** [comment]
Thanks @fr3fou ! would you mind rebasing the PR?

I haven't played with `dokku checks` so far, would you mind letting me know your use case for this module?

From [the docs](https://dokku.com/docs/deployment/zero-downtime-deploys/) I understand that zero-downtime checks are enabled on all apps per default.
Have you found it necessary to disable these checks for some apps?

**Comment by fr3fou** [comment]
I'll rebase it in an hour. As for my usecase - apps that have a published port can't be redeployed with checks enabled, so I'd have to disable them

File: library/dokku_checks.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+author: Simo Aleksandrov
+"""
+
+EXAMPLES = """
+- name: Enable the zero downtime deployment (enabled by default)
+  dokku_checks:
+    app: hello-world
+    state: present
+
+- name: Disable the zero downtime deployment
```
Comment (ltalirz) [inline library/dokku_checks.py:35]: Since this is the actual use case, let's move this example up

File: library/dokku_checks.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      - The state of the checks functionality
+    required: False
+    default: present
+    choices: [ "present", "absent" ]
+    aliases: []
+author: Simo Aleksandrov
+"""
+
+EXAMPLES = """
+- name: Enable the zero downtime deployment (enabled by default)
```
Comment (ltalirz) [inline library/dokku_checks.py:30]:
```suggestion
- name: Re-enable the zero downtime deployment (enabled by default)
```

File: molecule/default/verify.yml [hunk 3]
```diff
@@ -245,3 +245,21 @@
       msg: |-
         'false' not found in output of 'dokku http-auth:report':
         {{ dokku_http_auth_off.stdout }}
+
+  # Testing dokku_checks
+  - name: Enabling the Zero Downtime deployment
```
Comment (ltalirz) [inline molecule/default/verify.yml:250]: shouldn't we rather be testing disabling the checks then?

**Review by ltalirz — COMMENTED** [review]
thanks @fr3fou - the code looks good, perhaps we should modify the test?

File: molecule/default/verify.yml (hunk 3 above)
Comment (fr3fou) [reply molecule/default/verify.yml:250]: hmm, I had tests for both cases, guess I removed it accidentally when rebasing

File: molecule/default/verify.yml [hunk 4]
```diff
@@ -245,3 +245,38 @@
...
+  - name: Disabling the Zero Downtime deployment
+    dokku_checks:
+      app: example-app
+      state: absent
+
+  - name: Get checks output # noqa 301
+    command: dokku checks:report example-app
+    register: dokku_checks
+
+  - name: Check that the checks were enabled
```
Comment (ltalirz) [inline molecule/default/verify.yml:259]:
```suggestion
  - name: Check that the checks were disabled
```

**Review by ltalirz — APPROVED** [review]
thanks, will merge once the tests pass

I'll fix the word in rebasing my dokku_acl pr

**Pull request merged**